tests/window_management_tests/CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+include(CMakeDependentOption)
+
+include_directories(
+  ${CMAKE_SOURCE_DIR}
+  ${CMAKE_CURRENT_BINARY_DIR}
+  ${PROJECT_SOURCE_DIR}/src/include/platform
+  ${PROJECT_SOURCE_DIR}/src/include/cookie
+  ${PROJECT_SOURCE_DIR}/src/include/common
+  ${PROJECT_SOURCE_DIR}/src/include/server
+  ${PROJECT_SOURCE_DIR}/src/include/client
+  ${PROJECT_SOURCE_DIR}/src/include/gl
+  ${PROJECT_SOURCE_DIR}/include/renderers/sw
+)
+
+set(WINDOW_MANAGEMENT_TESTS_SRCS
+  test_minimal_window_manager.cpp
+)
+
+link_directories(${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+mir_add_wrapped_executable(mir_window_management_tests NOINSTALL
+  ${WINDOW_MANAGEMENT_TESTS_SRCS}
+  ${MIR_SERVER_OBJECTS}
+  ${MIR_PLATFORM_OBJECTS}
+)
+
+if (MIR_USE_PRECOMPILED_HEADERS)
+  target_precompile_headers(
+    mir_window_management_tests
+    PRIVATE
+    <gmock/gmock.h>
+    <gtest/gtest.h>
+    <vector>
+    <string>
+    <exception>
+    <memory>
+  )
+endif()
+
+add_dependencies(mir_window_management_tests GMock)
+
+target_link_libraries(
+  mir_window_management_tests
+
+  mir-test-static
+  mir-test-framework-static
+  mir-test-doubles-static
+
+  mircommon
+  miral
+  miral-internal
```

Comment:
Linking both `miral` (which includes `miral-internal` internally) and `miral-internal` risk multiple definitions. (Which is at best non-portable, and at worst causes hard to track down bugs.)

Reply:
See #3424